components/editableText/EditableText.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { SmallButton, Icon, Input } from 'react-components';
+import { c } from 'ttag';
+
+const EditableText = ({ icon, onSubmit, initialText, children, readOnly, className, ...rest }) => {
+    const [inputValue, setInputValue] = useState(initialText);
+    const [editing, setEditing] = useState(false);
+
+    useEffect(() => {
+        if (editing) {
+            setInputValue(initialText);
+        }
+    }, [editing, initialText]);
+
+    const onToggleEditing = () => {
+        setEditing(!editing);
+    };
+
+    const submit = (value) => {
+        onSubmit(value);
+        setEditing(false);
+    };
+
+    const handleFormSubmit = (e) => {
+        e.preventDefault();
+        submit(inputValue);
+    };
+
+    const handleChangeInputValue = ({ target }) => setInputValue(target.value);
+
+    return editing ? (
+        <form className="flex" onSubmit={handleFormSubmit}>
+            <SmallButton onClick={onToggleEditing} className="mr0-5" title={c('Action').t`Close`}>
+                <Icon name="close" />
+            </SmallButton>
+            {children ? (
+                children({ submit })
+            ) : (
+                <>
+                    <SmallButton type="submit" className="mr0-5" title={c('Action').t`Confirm`}>
```

Comment:
You probably need to use `pm-button--for-icon` class, [see here](https://design-system-beta.netlify.com/buttons/).